### PR TITLE
Add voter for workflow right

### PR DESCRIPTION
**Pull request by veolys-nanne**

### CODE REVIEW

**Comment by ngilain** [comment]
:+1:

File: BaseApiBundle/Controller/BaseController.php [hunk 1]
```diff
@@ -56,9 +57,22 @@ protected function reverseTransform(Request $request, $id, $type, $event, $event
...
         $mixed = $this->get('open_orchestra_model.repository.' . $typeName)->find($id);
+        $fromStatus = $mixed->getStatus();
         $mixed = $this->get('open_orchestra_api.transformer_manager')->get($typeName)->reverseTransform($facade, $mixed);
+        $toStatus = $mixed->getStatus();
 
-        if ($this->isValid($mixed)) {
+        $granted = true;
+        if ($fromStatus->getId() != $toStatus->getId()) {
+            $role = $this->get('open_orchestra_model.repository.role')->findOneByFromStatusAndToStatus($fromStatus, $toStatus);
+            $workflowFunctions = $this->get('open_orchestra_workflow_function.repository.workflow_function')->findByRole($role);
```
Comment (nicolasThal) [inline BaseApiBundle/Controller/BaseController.php:67]: is there a relation between base-api-bundle and workflow-bundle ?

File: BaseApiBundle/Controller/BaseController.php [hunk 2]
```diff
@@ -56,9 +57,22 @@ protected function reverseTransform(Request $request, $id, $type, $event, $event
...
+            $attributes = array();
+            foreach($workflowFunctions as $workflowFunction){
+                $attributes[] = $workflowFunction->getId();
+            }
+            $granted = $this->get('security.authorization_checker')->isGranted($attributes, $mixed);
```
Comment (nicolasThal) [inline BaseApiBundle/Controller/BaseController.php:72]: the is_granted should not be done here but in the transformer

Comment (veolys-nanne) [inline BaseApiBundle/Controller/BaseController.php:72]: That's the point i was talking about on skype ^^

**Pull request closed without merging**